src/library/searchquery.h

```diff
@@ -184,6 +184,52 @@ class DurationFilterNode : public NumericFilterNode {
     double parse(const QString& arg, bool* ok) override;
 };
 
+// BPM filter that supports fuzzy matching via ~ prefix.
+// If no operator is provided (bpm:123) it also finds half & double BPM matches.
+// Half/double values aren't integers, int ranges are used. E.g. bpm:123.1 finds
+// 61-61, 123.1 and 246-247 BPM
```

Comment:
```suggestion
// 61-62, 123.1 and 246-247 BPM
```

Reply:
Why it finds only a full match in the 1x case but a 1 BPM rang in other cases? I don't think that the underlying use case is different. 
Did you consider to always match integer rounded values? 

Or distinguish 123.0 for only 123.0 but 123 for 122.5 .. 123.5

Reply:
It's only a 1 BPM for half and double if the search term has decimals, because it that case it seems unlikely the user has/wants exact half/double matches.

Reply:
Yes I agree but why do the user want exact 1x matches but not exact 2x and 0.5x matched. 
You can easily have the same track annotate as 123 or 62,5. So it will be found if it is in the 0.5x range but not in the 1x range. That doe not feel reasonable. 

Did you consider to use this: 
* 123 find 122.5 .. 123.5 | 62 .. 63 | 245,5 .. 246,5 
* 123.3 find 123.25 .. 123.35 | 62,1 .. 63,2 | 246,55 .. 246,65
That means that the omitted decimal places are matching to the rounded value of the searched tracks.

Reply:
The comment is now outdated.